packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java

```diff
@@ -147,99 +123,53 @@ private void stop(boolean success) {
     }
   }
 
-  /**
-   * If the activity is paused or stopped, we have to stop listening for fingerprint. Otherwise,
-   * user can still interact with fp reader in the background.. Sigh..
-   */
   @Override
-  public void onActivityPaused(Activity activity) {
-    if (call.argument("stickyAuth")) {
-      pause();
-    } else {
+  public void onAuthenticationError(int errorCode, CharSequence errString) {
+    if (!activityPaused || !isAuthSticky) {
```

Comment:
optional, subjective: The negatives combined with the logical or makes this kind of hard to read. Can we rephrase this?

```java
if (activityPaused && isAuthSticky) {
  return;
}

// Either the authentication got cancelled by user or we are not interested
// in sticky auth, so return failure.
stop(false);
```

Reply:
Good suggestion. Applied.